### PR TITLE
add probe for citadel.

### PR DESCRIPTION
**Pull request by morvencao**

### CODE REVIEW

File: security/citadel/templates/deployment.yaml [hunk 1]
```diff
@@ -55,6 +55,12 @@ spec:
           {{- if .Values.security.trustDomain }}
             - --trust-domain={{ .Values.security.trustDomain }}
           {{- end }}
+          livenessProbe:
+            httpGet:
+              path: /version
+              port: {{ .Values.global.monitoringPort }}
```
Comment (sdake) [inline security/citadel/templates/deployment.yaml]: This is not in globals.yaml.  I believe we decided to hardcode the monitoring port for 1.2.

**Comment by sdake** [comment]
A grep of istio-installer shows 15014 port is hardcoded.

File: security/citadel/templates/deployment.yaml (hunk 1 above)
Comment (costinm) [reply security/citadel/templates/deployment.yaml]: Yes, some of the settings we never tested and likely don't support should be replaced by default values.

We support kustomize for cases user really needs a custom value.

**Comment by morvencao** [comment]
@sdake Thanks, replace probe port with hard coded value.

**Pull request merged**